cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -140,4 +140,22 @@ UTF8_LENGTH(char_length, utf8)
 UTF8_LENGTH(length, utf8)
 UTF8_LENGTH(lengthUtf8, binary)
 
+// Convert a utf8 sequence to upper case.
+FORCE_INLINE
+char* upper_utf8(int64 context, const char* data, int32 data_len, int32_t* out_len) {
```

Comment:
I suppose you know that, but the implementation is obviously wrong when it comes to non-ASCII characters. For example in Python:
```
>>> "é".upper()
'É'
```

Reply:
I'll check on the expected sql behavior for this.

Reply:
On Sqlite, dremio, impala - the "é" remains unchanged (i.e they handle only ascii characters). On Postgres, the  "é" changes to 'É'.

I'm leaving a TODO here. because, modifying it is non-trivial and depends on the locale.

Reply:
You wouldn't want to depend on the locale. It's UTF8, not some unknown encoding ;-) But, yes, unicode case manipulation is complicated and you may be better off using an external library like ICU (you could still have a fast path for ASCII).

I'm fine with leaving a TODO note or something.